rebar.config

```diff
@@ -1,8 +1,31 @@
 {erl_opts, [debug_info]}.
-{project_plugins, [rebar3_proper]}.
 {deps, []}.
-{profiles,
-    [{test, [
-        {deps, [{proper, "1.2.0"}]}
+{project_plugins, [
+    rebar3_proper,
+    {rebar3_gpb_plugin, "1.12.1"}
+]}.
+{gpb_opts, [
+    {i, "proto"},
+    {o_erl, "src"},
+    {o_hrl, "src"},
+    {verify, always},
+    {strings_as_binaries, true},
+    {maps, true},
+    {maps_unset_optional, omitted},
+    {report_warnings, true},
+    {target_erlang_version, 17}
```

Comment:
@ericmj fyi I've run into some build issues on previous configuration: https://travis-ci.org/hexpm/hex/jobs/351926875

I've first tried solving it by pinning to earlier gpb plugin version (which uses gpb 3.x) but that didn't solve it so I changed targeted otp version and we're good. 

In https://github.com/hexpm/specifications/blob/master/registry-v2.md#formats we still mention gpb 3.x so I guess we can leave this as is but there's no problem with updating to gpb 4.x too.